[Teo Ming Huang] iP

teominghuang opened the pull request:
# DockerHawker

> "Your mind is for having ideas, not holding them." -- David Allen [(source)](https://dansilvestre.com/productivity-quotes)

DockerHawker frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😜 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

teominghuang committed:
Level 1 - 50%

teominghuang committed:
Level 1 - Completed

teominghuang committed:
Level 1 - Editted

teominghuang committed:
Level 1 - Edit2

teominghuang committed:
Level 2 - Completed

teominghuang committed:
Mark Feature Completed

teominghuang committed:
Edits for level 3

teominghuang committed:
Level 3 & Unmark Completed

teominghuang committed:
Major refactoring and changing of data structure

teominghuang committed:
Editting of Input.txt & EXPECTED.txt

teominghuang committed:
2 Error-handling examples handled

teominghuang committed:
More comprehensive automated testings

teominghuang committed:
Developed better testings

teominghuang committed:
Deletion feature integrated

teominghuang committed:
Improved Testing

teominghuang committed:
Editted comments

teominghuang committed:
Create database, edit list, and create task feature

teominghuang committed:
Connect to txt file, Add CRUD features

teominghuang committed:
Add LocalDate attribute & Search deadlines on specific date feature

teominghuang committed:
Merge branch 'branch-Level-7' into master

teominghuang committed:
merge Level-8

michaelseyo commented on src/main/java/task/Tasks.java:
```diff
@@ -0,0 +1,57 @@
+        this.completed = false;
+    }
+
+    public Tasks(String taskName, boolean completion) {
```
I like the use of the overloaded constructor here, neat trick :)

michaelseyo commented on src/main/java/task/Events.java:
```diff
@@ -0,0 +1,30 @@
+    }
+    //toString returning event
+    public String toString(){
+        return "[E][" + (this.getCompletion()==true?"X":" ") + "] " + super.getName() 
```
Perhaps we could declare a function to generate the String you're returning so that it'll be clearer and we can avoid 'magic' strings?

michaelseyo commented on src/main/java/command/Commands.java:
```diff
@@ -0,0 +1,6 @@
+package command;
+
+//Enums of command
+public enum Commands {
```
Nice use of enums for Commands which we will definitely keep on using throughout the building of this project!

michaelseyo commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,17 @@
+blah
+todo
+event
+deadline
```
Great that you're considering the case for empty description!

michaelseyo commented on src/main/java/task/Deadlines.java:
```diff
@@ -0,0 +1,37 @@
+
+    // Save to database format
+    public String toDatabaseString() {
+        return "D | " + (this.getCompletion() == true ? "X" : " ") + " | " + super.getName()
+                + " | " + deadline + "\n";
```
Perhaps we could simplify this expression further? For example maybe we could let "D | " be a static final variable that is named, such that the overall expression is clearer

michaelseyo reviewed (commented):
Overall everything looks great, just some simplifying to do so that the code is clearer

teominghuang committed:
Turn project OOP-based

teominghuang committed:
Organise code into packages

teominghuang committed:
Package data and key information into packages

teominghuang committed:
Package data and key information into packages

teominghuang committed:
create duke package and refactor files

teominghuang committed:
add jUnit tests

teominghuang committed:
create java jar file for program

teominghuang committed:
Add javadoc comments

teominghuang committed:
Edit to fit CS2103T coding conventions

teominghuang committed:
Complete Find Command

teominghuang committed:
Merge branch 'branch-A-JavaDoc'

teominghuang committed:
Resolve merge conflicts

teominghuang committed:
Create FindCommand, Resolve merge conflicts

teominghuang committed:
Fix some minor bugs

KristopherPTaslim commented on src/main/java/duke/task/Deadlines.java:
```diff
@@ -0,0 +1,65 @@
+ * name, completion status, and the deadline of the event e.g.,
+ * <code>"Eat lunch", true, "2012-06-03"</code>
+ */
+public class Deadlines extends Tasks {
```
Small thing, but perhaps you could change Deadlines to deadline cause it is referring to a single Task

albertsutz replied:
I agree.

KristopherPTaslim commented on src/main/java/duke/task/Events.java:
```diff
@@ -0,0 +1,59 @@
+ * name, completion status, and timing of the event e.g.,
+ * <code>"Eat lunch", true, "9am"</code>
+ */
+public class Events extends Tasks {
```
similar to deadline

KristopherPTaslim commented on src/main/java/duke/task/Tasks.java:
```diff
@@ -0,0 +1,69 @@
+ * A <code>Tasks</code> is represented by its name and completion status e.g.,
+ * <code>"Eat breakfast by 9am", true</code>
+ */
+public class Tasks {
```
similar as above

KristopherPTaslim commented on src/main/java/duke/task/Todos.java:
```diff
@@ -0,0 +1,50 @@
+ * A <code>Todo</code> inherits from <code>Tasks</code> and is represented by its
+ * name, and completion status e.g.,<code>"Eat dinner", true</code>
+ */
+public class Todos extends Tasks {
```
similar as above

KristopherPTaslim reviewed (commented):
Overall, very nice coding style and clean, just a very very minor mistake. Goodluck on the future Ips!!

albertsutz commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,93 @@
+import duke.task.Events;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
```
Perhaps you can separate lines 9 and 10 to differentiate between the duke and java package imports.

albertsutz commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,93 @@
+
+    private static final boolean IS_EXIT = false;
+    private final String commandWord;
+    private final String arguments; // In the form of user duke.command
```
Not sure whether an inline comment violates the coding standard or not. Perhaps you can put the comments above the attribute declaration?

albertsutz commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,93 @@
+    private static final boolean IS_EXIT = false;
+    private final String commandWord;
+    private final String arguments; // In the form of user duke.command
+
```
Don't forget to put Javadoc for `AddCommand` and `execute`! I believe the rest of the files are already filled with the appropriate javadocs.

albertsutz replied:
I see that the same methods on other files are also not documented using JavaDocs. I would recommend you to look add the appropriate JavaDocs for completeness.

albertsutz commented on src/main/java/duke/command/Commands.java:
```diff
@@ -0,0 +1,44 @@
+ * blueprint for creation of future commands.
+ */
+public class Commands {
+
```
This is just me being a bit picky, but maybe you can remove the Whiteline here.

albertsutz commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,55 @@
+    public static final String FAILURE_MESSAGE = "    'Delete' Command Executed Unsuccessfully";
+
+    private static boolean IS_EXIT = false;
+    private String arguments; // In the form of user duke.command
```
Similar to the previous commend, perhaps you can put the commends above the statement?

albertsutz commented on src/main/java/duke/data/DukeDatabase.txt:
```diff
@@ -0,0 +1,3 @@
+T |   | hello my name is ben
+E | X | hello you are so cool | 2012-03-03
+T |   | Ben is my name
```
Hi, I think that the database file should not be included in the Git repository. However, if you want to make this file as a default initial database, it can also work fine!

albertsutz commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,70 @@
+        }
+
+        switch (commandWord) {
+
```
This is me being picky again, but maybe you could remove some of the blank lines to increase readability? However, in terms of code, everything looks good for the switch case.

albertsutz commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,134 @@
+     * If the method is successful, it will return an arraylist with <code>Tasks</code>,
+     * else it returns one that is empty.
+     *
+     * @return an arraylist with all the Tasks stored in the database.
```
Don't forget to use capital letter for "An arraylist with all the ..."!

I noticed that some of the other javaDocs documentation lines also don't start with a capital letter. Perhaps you can look into this.

albertsutz commented on src/main/java/duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,186 @@
+    public String queryTasks(String query) {
+        ArrayList<Tasks> unfilteredTasks = new ArrayList<Tasks>(taskList);
+        ArrayList<Tasks> filteredTasks = new ArrayList<Tasks>();
+        StringBuilder sb1 = new StringBuilder("    Here are the matching tasks in your list: \n");
```
Perhaps its better to not use number (`sb1`, `sb2`, etc) as discussed in the lecture video.

albertsutz reviewed (commented):
Overall, really good job on implementing Duke! I can definitely see your way of thinking and there are not many problems in the coding style aspects. However, I would highly suggest that you perhaps look into some of my review and maybe it could make your code more complete.

Some overall things are:

- You are missing some of the javadocs comments for several public methods
- there are a lot of common constants (like `IS_EXIT`) which I think can be abstracted and put into a new Constants Class.

Good job and good luck for the next increments!

teominghuang committed:
tweak database

teominghuang committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

teominghuang committed:
merge with gradle-support

teominghuang committed:
Automate project build with Gradle

teominghuang committed:
Debug run gradlew, add javadocs

teominghuang committed:
Merge branch 'branch-A-Gradle'

teominghuang committed:
Adhere to module coding standards

teominghuang committed:
Merge branch 'branch-A-CheckStyle'

teominghuang committed:
Create GUI

teominghuang committed:
Fix GUI bugs

teominghuang committed:
Merge branch 'branch-Level-10'

teominghuang committed:
Refactor code based on guide

teominghuang committed:
Add assertion statements

teominghuang committed:
Merge pull request #2 from teominghuang/branch-A-Assertions

Add assertion statements

teominghuang committed:
Fix merge conflicts

teominghuang committed:
Merge pull request #3 from teominghuang/branch-A-CodeQuality

Refactor code based on guide

teominghuang committed:
Merge branch 'master' of https://github.com/teominghuang/ip

teominghuang committed:
Edit the delete command to allow for mass deletion

Delete allow for only deletion of one task at a time

A mass deletion method is created and integrated into the delete
task method.

A mass deletion method allows for greater convenience when the user
need to delete multiple tasks.

teominghuang committed:
Merge branch 'branch-BCD-Extension'

teominghuang committed:
JUnit testing: Created new JUnit tests for a more comprehensive test

Several refactoring were made previously, and tests and stubs
are unable to perform a comprehensive test on the system.

Several JUnit tests were created, specifically for the smaller
methods. These tests are important as it is used widely across
the code base.

It is done this way to improve the comprehensiveness of the testing
unit.

Let's,
* think of better ways to increase the scope of the other JUnit
tests.

teominghuang committed:
Improve GUI

teominghuang committed:
Add GUI Features

teominghuang committed:
Add javadoc to most public methods

teominghuang committed:
Merge branch 'branch-A-BetterGUI'

teominghuang committed:
Fix bugs, Add more GUI

teominghuang committed:
Add User Guide

teominghuang committed:
final-fixes

teominghuang committed:
Set theme jekyll-theme-minimal

teominghuang committed:
Set theme jekyll-theme-slate

teominghuang committed:
Fix a single bug

teominghuang committed:
Edit Storage

Storage used absolute paths to find and create a database. It is
changed to relative pathing so that users can run the program
anywhere and it should work.

teominghuang committed:
Merge branch 'branch-FixingJavaFXBugs'

teominghuang committed:
Set theme jekyll-theme-slate

teominghuang committed:
Solve BugFixes

teominghuang committed:
Merge branch 'master' of https://github.com/teominghuang/ip
